src/libraries/System.Speech/src/AudioFormat/AudioFormatConverter.cs

```diff
@@ -0,0 +1,300 @@
+            Text,
+            NonStandardFormat,     // Non-SAPI 5.1 standard format with no WAVEFORMATEX description
+            ExtendedAudioFormat,   // Non-SAPI 5.1 standard format but has WAVEFORMATEX description
+            // Standard PCM wave formats
+            PCM_8kHz8BitMono,
+            PCM_8kHz8BitStereo,
+            PCM_8kHz16BitMono,
+            PCM_8kHz16BitStereo,
+            PCM_11kHz8BitMono,
+            PCM_11kHz8BitStereo,
+            PCM_11kHz16BitMono,
+            PCM_11kHz16BitStereo,
+            PCM_12kHz8BitMono,
+            PCM_12kHz8BitStereo,
+            PCM_12kHz16BitMono,
+            PCM_12kHz16BitStereo,
+            PCM_16kHz8BitMono,
+            PCM_16kHz8BitStereo,
+            PCM_16kHz16BitMono,
+            PCM_16kHz16BitStereo,
+            PCM_22kHz8BitMono,
+            PCM_22kHz8BitStereo,
+            PCM_22kHz16BitMono,
+            PCM_22kHz16BitStereo,
+            PCM_24kHz8BitMono,
+            PCM_24kHz8BitStereo,
+            PCM_24kHz16BitMono,
+            PCM_24kHz16BitStereo,
+            PCM_32kHz8BitMono,
+            PCM_32kHz8BitStereo,
+            PCM_32kHz16BitMono,
+            PCM_32kHz16BitStereo,
+            PCM_44kHz8BitMono,
+            PCM_44kHz8BitStereo,
+            PCM_44kHz16BitMono,
+            PCM_44kHz16BitStereo,
+            PCM_48kHz8BitMono,
+            PCM_48kHz8BitStereo,
+            PCM_48kHz16BitMono,
+            PCM_48kHz16BitStereo,
+            // TrueSpeech format
+
+            TrueSpeech_8kHz1BitMono,
+            // A-Law formats
+            CCITT_ALaw_8kHzMono,
+            CCITT_ALaw_8kHzStereo,
+            CCITT_ALaw_11kHzMono,
+            CCITT_ALaw_11kHzStereo,
+            CCITT_ALaw_22kHzMono,
+            CCITT_ALaw_22kHzStereo,
+            CCITT_ALaw_44kHzMono,
+            CCITT_ALaw_44kHzStereo,
+            // u-Law formats
+            CCITT_uLaw_8kHzMono,
+            CCITT_uLaw_8kHzStereo,
+            CCITT_uLaw_11kHzMono,
+            CCITT_uLaw_11kHzStereo,
+            CCITT_uLaw_22kHzMono,
+            CCITT_uLaw_22kHzStereo,
+            CCITT_uLaw_44kHzMono,
+            CCITT_uLaw_44kHzStereo,
+            // ADPCM formats
+            ADPCM_8kHzMono,
+            ADPCM_8kHzStereo,
+            ADPCM_11kHzMono,
+            ADPCM_11kHzStereo,
+            ADPCM_22kHzMono,
+            ADPCM_22kHzStereo,
+            ADPCM_44kHzMono,
+            ADPCM_44kHzStereo,
+            // GSM 6.10 formats
+            GSM610_8kHzMono,
+            GSM610_11kHzMono,
+            GSM610_22kHzMono,
+            GSM610_44kHzMono,
+            NUM_FORMATS
+        }
+
+        #endregion
+
+        #region Private Type
+
+        private enum WaveFormatId
+        {
+            Pcm = 1,
+            AdPcm = 0x0002,
+            TrueSpeech = 0x0022,
+            Alaw = 0x0006,
+            Mulaw = 0x0007,
+            Gsm610 = 0x0031
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
```

Comment:
Why not make this a struct instead?

Reply:
I'd rather not make that kind of change to the original code. There's lots I would improve if this code was going to be worked on long term.